ic-factory/src/state.rs

```diff
@@ -102,19 +102,8 @@ impl FactoryState {
     /// Returns `FactoryError::AccessDenied` if the caller is not the factory controller.
     pub fn authorize_owner(&mut self) -> Result<Authorized<Owner>, FactoryError> {
         let caller = ic_canister::ic_kit::ic::caller();
-        self.authorize_owner_int(caller)
-    }
-
-    // In some cases we need to call owner-only operations after `await` point, when `ic::caller()`
-    // cannot be called (due to IC limitation). In this case we use this method with the caller,
-    // retrieved before the `await` point. To prevent incorrect usage of this method it is
-    // `pub(crate)` only.
-    pub(crate) fn authorize_owner_int<'a>(
-        &'a mut self,
-        caller: Principal,
-    ) -> Result<Authorized<'a, Owner>, FactoryError> {
         if caller == self.configuration.controller {
-            Ok(Authorized::<'a, Owner> {
+            Ok(Authorized::<Owner> {
```

Comment:
`Authorized` has a lifetime so we should always supply one, however since it's elided (because of `&mut self`) we can provide an anonymous lifetime.
This applies to the return type as well
```suggestion
            Ok(Authorized::<'_, Owner> {
```

Reply:
If it follows elision rules, why should we provide an anonymous lifetime then?